presto-orc/src/main/java/com/facebook/presto/orc/writer/StreamOrderingLayout.java

```diff
@@ -0,0 +1,212 @@
+                    }
+                }
+            }
+        }
+        return new StreamMetadata(sequenceToStreamsBuilder.build(), keyToSequenceBuilder.build());
+    }
+
+    @Override
+    public void reorder(
+            List<StreamDataOutput> dataStreams,
+            Map<Integer, Integer> nodeIdToColumn,
+            Map<Integer, ColumnEncoding> nodeIdToColumnEncodings)
+    {
+        List<StreamDataOutput> nonReorderStreams = new ArrayList<>();
+        StreamMetadata metadata = getStreamMetadata(nodeIdToColumn, nodeIdToColumnEncodings, config);
+        Map<ColumnSequenceInfo, List<StreamDataOutput>> sequenceToStreams = metadata.sequenceToStreams;
+        for (StreamDataOutput dataOutput : dataStreams) {
+            int nodeId = dataOutput.getStream().getColumn();
+            int sequence = dataOutput.getStream().getSequence();
+            int column = nodeIdToColumn.get(nodeId);
+            // only if sequence ID > 0, we do a look up in sequenceToStreams
+            if (sequence > 0) {
+                List<StreamDataOutput> streams = sequenceToStreams.get(new ColumnSequenceInfo(column, sequence));
+                if (streams == null) {
+                    nonReorderStreams.add(dataOutput);
+                }
+                else {
+                    streams.add(dataOutput);
+                }
+            }
+            else {
+                nonReorderStreams.add(dataOutput);
+            }
+        }
+
+        // reorder everything in the input order
+        List<StreamDataOutput> orderedStreams = new ArrayList<>();
+        Map<ColumnKeyInfo, Integer> keyToSequence = metadata.keyToSequence;
+        for (Map.Entry<Integer, List<DwrfProto.KeyInfo>> columnToKeys : config.getStreamOrdering().entrySet()) {
+            int column = columnToKeys.getKey();
+            for (DwrfProto.KeyInfo key : columnToKeys.getValue()) {
+                ColumnKeyInfo columnKeyInfo = new ColumnKeyInfo(column, key);
+                Integer sequence = keyToSequence.get(columnKeyInfo);
+                if (sequence != null) {
+                    ColumnSequenceInfo columnSequenceInfo = new ColumnSequenceInfo(column, sequence);
+                    List<StreamDataOutput> groupedDataStreams = sequenceToStreams.get(columnSequenceInfo);
+                    checkState(groupedDataStreams != null, "list of streams for a sequence cannot be null");
+                    checkState(groupedDataStreams.size() > 0, "There should be at least one stream for a sequence");
+                    orderedStreams.addAll(groupedDataStreams);
+                }
+            }
+        }
+
+        // do actual reordering
+        nonStreamOrderingLayout.reorder(nonReorderStreams, ImmutableMap.of(), ImmutableMap.of());
+
+        // add all the streams
+        checkState(orderedStreams.size() + nonReorderStreams.size() == dataStreams.size(),
+                String.format(
```

Comment:
Don't use `String.format`. Instead use the `checkState`'s `%s` placeholder for arguments. 

```
checkState(condition, "bla %s, bla %s", orderedStreams.size(), nonReorderStreams.size());
```

Don't use anything else other than `%s` with the Preconditions, it is not supported.